### PR TITLE
Cap `learn_rounding_scheme` correctly

### PR DESCRIPTION
**Pull request by fealho**
Fix code analogous to [#RDT/556](https://github.com/sdv-dev/RDT/issues/556).

### CODE REVIEW

**Review by amontanez24 — APPROVED** [review]
Good catch!

File: sdv/data_processing/numerical_formatter.py [hunk 1]
```diff
@@ -48,15 +48,25 @@ def __init__(self, learn_rounding_scheme=False, enforce_min_max_values=False,
...
+        if ((roundable_data % 1) == 0).all():
+            return 0
+
+        # Try to round to fewer digits
+        if (roundable_data == roundable_data.round(MAX_DECIMALS)).all():
+            for decimal in range(MAX_DECIMALS + 1):
+                if (roundable_data == roundable_data.round(decimal)).all():
+                    return decimal
+
+        # Can't round, not equal after MAX_DECIMALS digits of precision
```
Comment (amontanez24) [inline sdv/data_processing/numerical_formatter.py:69]: Can we raise a warning here that informs the user that the rounding could not be detected? Something like ```No rounding scheme detected for column 'interest_rate'. Synthetic data will not be rounded.```

Comment (fealho) [reply sdv/data_processing/numerical_formatter.py:69]: Good catch, I completely forgot about the warning message... thanks for reminding me!

**Review by amontanez24 — APPROVED** [review]
LGTM! :shipit:

**Review by pvk-developer — APPROVED** [review]

**Pull request merged**